### ISSUE TITLE
Incorrect value when returned promise is resolved asynchronously

### ISSUE THREAD

**Issue by mickylad** [charleshansen/mock-promises#23]
Hi,

There seems to be a bug with returning promises in a chain that are resolved asynchronously.

When a returned promise is resolved synchronously, the chained promise is correctly resolved with it's value. However, when the returned promise is resolved asynchronously, the chained promise is resolved with undefined.

I've written some snippets to illustrate the difference:

```
const testedPromise = new Promise((resolve) => {
    resolve(1);
}).then(() =>
    new Promise((resolve) => {
        resolve(2);
    })
);

mockPromises.tick(2);

mockPromises.valueForPromise(testedPromise); // 2
```

```
let resolve2;

const testedPromise = new Promise((resolve) => {
    resolve(1);
}).then(() =>
    new Promise((resolve) => {
        resolve2 = resolve;
    })
);

mockPromises.tick(1);

resolve2(2);

mockPromises.tick(2);

mockPromises.valueForPromise(testedPromise); // undefined
```

I'm not sure exactly what the fix might be, I suspect there is an issue with the `resolvePromise` function in the way it handles `value` being a promise (which is the returned promise in these cases).

**Comment by charleshansen** [comment]
Reproduced. Weirdly, even the first example fails for Q promises. I'll think about a fix. I hoped this was solved by my fix to #22, but I guess not.

**Comment by charleshansen** [comment]
Sorry that took so long, I think I just fixed it in 0.8. Let me know if it works
